gm4_rope_ladders/data/gm4_rope_ladders/functions/mechanics/ladder_placement/place_ladder.mcfunction

```diff
@@ -12,6 +12,10 @@ execute if entity @s[gamemode=!creative] run item modify entity @s weapon.mainha
 # grant advancement
 advancement grant @s only gm4:rope_ladders
 
+# break block
+execute unless block ~ ~ ~ #gm4:water run setblock ~ ~ ~ air destroy
+execute if block ~ ~ ~ #gm4:water run setblock ~ ~ ~ water destroy
+
 # place ladder
-execute unless block ~ ~ ~ water run clone ~ ~1 ~ ~ ~1 ~ ~ ~ ~
-execute if block ~ ~ ~ water run function gm4_rope_ladders:mechanics/ladder_placement/place_water_ladder
+execute unless block ~ ~ ~ #gm4:water run clone ~ ~1 ~ ~ ~1 ~ ~ ~ ~
```

Comment:
technically this means that if the ladder above is waterlogged it will incorrectly copy a waterlogged ladder. However the only method I found to do this is by replacing light blocks (as they are replaceable but stop water flow), so it's not a big deal.